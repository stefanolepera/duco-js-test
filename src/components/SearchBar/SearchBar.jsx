import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledInput = styled.input`
    border: none;
    padding: 0 10px;
    margin: 0 auto;
    font-size: 1.2em;
    width: 600px;
    height: 40px;
    color: #000;
    background: #ccc;
    border-radius: 4px;

    &:focus {
        outline: none;
        box-shadow: 0 0 0 2px maroon;
    }
`;

class SearchBar extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            character: ''
        };
    }

    onChange = e => {
        this.setState({ [e.target.name]: e.target.value }, () => {
            this.props.handleOnChange(this.state.character);
        });
    };

    render() {
        const { character } = this.state;
        return (
            <StyledInput
                type="text"
                name="character"
                placeholder="Search Character"
                onChange={this.onChange}
                value={character}
                autoFocus
            />
        );
    }
}

SearchBar.propTypes = {
    handleOnChange: PropTypes.func
};

export default SearchBar;
