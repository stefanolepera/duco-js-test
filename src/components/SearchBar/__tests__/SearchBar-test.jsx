import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SearchBar from '../SearchBar';

describe('<ResultItem /> test', () => {
    it('Test snapshots of content', () => {
        const wrapper = shallow(<SearchBar />);

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
