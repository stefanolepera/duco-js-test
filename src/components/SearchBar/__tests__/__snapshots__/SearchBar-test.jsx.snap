// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultItem /> test Test snapshots of content 1`] = `
<styled.input
  autoFocus={true}
  name="character"
  onChange={[Function]}
  placeholder="Search Character"
  type="text"
  value=""
/>
`;
